### PR TITLE
less privs

### PR DESCRIPTION
**Pull request by till**
- Update: run kratos-admin-ui without root
- Chore: build test image on PR
- Chore: clean-up workflows

### CODE REVIEW

**Comment by dfoxg** [comment]
@till thanks for the pr! Could you remove the dependabot-config? I'm already using mend renovate for updating the packages. Thanks!

File: .github/workflows/build-on-pr.yml [hunk 1]
```diff
@@ -1,20 +1,29 @@
 name: Build Docker Image
 
 on:
-  push:
-    branches: 
-      - '*'
-      - '!main'
```
Comment (dfoxg) [inline .github/workflows/build-on-pr.yml:7]: Why did you remove the main exclusion?

File: .github/workflows/build-on-pr.yml [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
-        cd kratos-admin-ui
-        docker build .
\ No newline at end of file
+    - uses: docker/setup-buildx-action@v2
+    - uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - id: meta
```
Comment (dfoxg) [inline .github/workflows/build-on-pr.yml:20]: What's the benefit of using this actions instead of running the command directly?

**Comment by till** [comment]
> @till thanks for the pr! Could you remove the dependabot-config? I'm already using mend renovate for updating the packages. Thanks!

I didn't know if renovate updated the github actions that are used in workflows. It didn't look like it. But I can remove that if you don't like it.

File: .github/workflows/build-on-pr.yml [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
-      - '!main'
   pull_request:
-    branches: [ '*' ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
```
Comment (till) [inline .github/workflows/build-on-pr.yml:11]: This enables GHA's to write to your registry. So you don't need to maintain the token you have in the secret.

**Review by dfoxg — APPROVED** [review]
Thanks for the contribution :)

**Comment by dfoxg** [comment]
The docker-push fails at the moment - could you have a look at it?

**Review by dfoxg — CHANGES_REQUESTED** [review]
Build fails

**Comment by dfoxg** [comment]
closed with version 2.2.0 https://github.com/dfoxg/kratos-admin-ui/releases/tag/v2.2.0

**Pull request closed without merging**